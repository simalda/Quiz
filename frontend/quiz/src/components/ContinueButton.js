import React from "react";
 

class ContinueButton extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    let clasname =  "top nextQuestion button  "+ ((this.props.isAnswerSelected === 0) ? "gray" : "blue");
     
    return  <button
    className = {clasname}
    key="button"
    id="ContinueButton"
    onClick={() => this.props.nextTurn()}
  >
    NEXT QUESTION
  </button>
  }
}

export default ContinueButton;
