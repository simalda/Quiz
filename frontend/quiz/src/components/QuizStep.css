.answers{
    display:grid;
    width: 100%;
    
    
  }
  .notSelected{
    background-color:rgb(255, 255,255,0.3);
  }

  .nextQuestion{
    font-weight: bold;
    width: 100%;
    
    
  }