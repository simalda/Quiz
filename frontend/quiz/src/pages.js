const pages = {
    Login: "Login",//-1
    NumberofQuestions:"NumberofQuestions",//3
    Main: "Main",//1
    QuizStep: "QuizStep",//0
    Results:"Results",//2
    Statistics: "Statistics",//5
    AddQuestion:"AddQuestion",
    AboutUs: "AboutUs"
  };
  
  export default pages