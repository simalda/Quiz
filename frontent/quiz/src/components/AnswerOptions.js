import React, { Component } from "react";
import ReactDOM from "react-dom";
import "../index.css";

function AnswerOptions({ title, highlight, onClick }) {
  var clasname = "answer " + (!!highlight ? highlight : "");

  return (
    <div id="answer" className={clasname} onClick={() => onClick(title)}>
      <h4>{title}</h4>
    </div>
  );
}

export default AnswerOptions;
