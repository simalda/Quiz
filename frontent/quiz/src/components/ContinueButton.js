import React from 'react'


function ContinueButton({nextTurn}){
    return(<button key="button"  id="ContinueButton" onClick = {nextTurn} > Continue</button>)
  }

  export default  ContinueButton 