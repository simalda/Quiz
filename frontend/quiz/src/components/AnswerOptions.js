import React, { Component } from "react";
import ReactDOM from "react-dom";
 
import "./QuizStep";

function AnswerOptions({ title, highlight, onClick }) {
  var clasname =  "top10 answers button  "+ (!!highlight ? highlight : "notSelected");

  return (
    <div id="answer" className={clasname} onClick={() => onClick(title)}>
      <h4>{title}</h4>
    </div>
  );
}

export default AnswerOptions;
