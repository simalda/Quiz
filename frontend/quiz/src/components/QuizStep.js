import React  from "react";
import ReactDOM from "react-dom";
import AnswerOptions from "./AnswerOptions";
import ContinueButton from "./ContinueButton";

class QuizStep extends React.Component {
  constructor(props) {
    super(props);
  }


  renderAnswer(title) {
    return (
      <AnswerOptions
        title={title}
        key={title}
        highlight={
          title === this.props.selectedAnswer
            ? title === this.props.correctAnswer
              ? "correct"
              : "incorrect"
            : undefined
        }
        onClick={this.props.onAnswerSelected}
      />
    );
  }

  render() {
    return (
      <div>         
            <div className="row">
              <div className="col-lg-4"></div>
              <div className="col-lg-4">  
                  <div   className = "heade">{this.props.numberOfCurrentQuestion}\{this.props.numberOfQuestions}</div>
                  <div className ="text"><h3 id={"question"+this.props.numberOfCurrentQuestion}>{this.props.question}</h3></div>
                  <div className="">
                    {this.props.answerOptions.map(title =>
                      this.renderAnswer(title[0])
                    )}
                    <ContinueButton isAnswerSelected = {this.props.isAnswerSelected} nextTurn={() => this.props.continueButtonClicked()} />               
                  </div>
             </div>
               <div className="col-lg-4"></div>            
                         </div>
          </div>
                 );
  }
}

export default QuizStep;
