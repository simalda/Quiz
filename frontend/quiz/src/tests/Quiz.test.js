import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {mount, shallow, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import QuizStep from '../components/QuizStep';
import AnswerOptions from "../components/AnswerOptions";
import pages from "../pages";
import ContinueButton from "../components/ContinueButton";

Enzyme.configure({ adapter: new Adapter() });



const state ={
  quiz:[{
    quesion:"a",
    answer:["c","d","s"],
    correctAnswer: "cor"
    }  ],
  highlight:'none',
  page:pages.Main,
  numberOfCurrentQuestion:1,
  question:"a",
    answerOptions:["c","d","s","a"]
}

describe("Quiz step", () => {
  let wrapper;
  let mockSubmit;
  beforeEach(() => {
    mockSubmit = jest.fn();
    wrapper = shallow(<QuizStep {...state} submit={mockSubmit} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("State page is correct", () => {
    expect(state.page).toBe(pages.Main);
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<QuizStep {...state} />, div);
  });

  it('should show text', ()=>{
    const text = wrapper.find({ id: "question1" });
    expect(text.text()).toBe(state.question);
  })

  // it('should find one button for guest', () => {     
  //   const butto = wrapper.find({id: "answer"})
  //      expect(butto).toHaveLength(4);
  //       });  

        it('should find continue button', () => {     
          const butto = wrapper.find('ContinueButton')
             expect(butto).toHaveLength(1);
              });        
});



/*------------------------------------------------------------------------------------*/
// describe('Index', () => {
//   it('should render correctly', () => {
//     const output = shallow(
//       <LogInPage   />
//     );
//     expect(shallowToJson(output)).toMatchSnapshot();
//   });
// });

// describe("Quiz", () =>{
// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<QuizStep {...state} onAnswerSelected={()=>{}}/>, div);
  
// });
// });

// describe("When no answer has been selected", () =>{
//   let wrapper;
//   beforeAll(()=> {
//     wrapper = mount(<QuizStep {...state} nAnswerSelected={()=>{}}/>);
//   });

//   it('should have no background color', () => {
//       expect(wrapper.find("div.row.turn").props().style.backgroundColor).toBe('');     
//   });
//   });
  
// describe("When the wrong answer has been selected", () =>{
//   let wrapper;
//   beforeAll(()=> {
//     wrapper = mount(<QuizStep {...(Object.assign({},state,{highlight: 'wrong'}))} nAnswerSelected={()=>{}}/>);
//   });

//   it('should have red background color', () => {
//       expect(wrapper.find("div.row.turn").props().style.backgroundColor).toBe('red');     
//   });
//   });

// describe("When the correct answer has been selected", () =>{
//     let wrapper;
//     beforeAll(()=> {
//       wrapper = mount(<QuizStep {...(Object.assign({},state,{highlight: 'correct'}))} nAnswerSelected={()=>{}}/>);
//     });
  
//     it('should have red background color', () => {
//         expect(wrapper.find("div.row.turn").props().style.backgroundColor).toBe('green');     
//     });
//     });

// describe("When the first answer is selected", () =>{
//   let wrapper;
//   const handleAnswersSelecter = jest.fn();

//   beforeAll(()=> {
//     wrapper = mount(<QuizStep {...state} onAnswerSelected={()=>{}}/>);
//     wrapper.find.apply('.kAnswer').first().simulate('click');
//   });

//   it('onAnswerselected should be called', () => {
//       expect(handleAnswersSelecter).toHaveBeenCalled();     
//   });

//   it("should receive c", ()=>{
//     expect(handleAnswersSelecter).toHaveBeenCalledWith("c");
//   })
//   });
    