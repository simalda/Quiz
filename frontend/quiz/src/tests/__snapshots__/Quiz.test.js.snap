// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz step should match the snapshot 1`] = `ShallowWrapper {}`;
